src/query/storage/index.go

```diff
@@ -66,8 +67,11 @@ func FromIdentTagIteratorToTags(identTags ident.TagIterator) (models.Tags, error
 func TagsToIdentTagIterator(tags models.Tags) ident.TagIterator {
 	identTags := make([]ident.Tag, 0, len(tags))
 
+	//TODO get a checked bytes pool here instead
 	for _, t := range tags {
-		identTags = append(identTags, ident.StringTag(t.Name, t.Value))
+		n := checked.NewBytes(t.Name, checked.NewBytesOptions())
+		v := checked.NewBytes(t.Value, checked.NewBytesOptions())
+		identTags = append(identTags, ident.BinaryTag(n, v))
```

Comment:
For now, perhaps we should replace with just a zero copy version (since models.Tags should be coming from the wire and won't be mutated?):
```go
identTags := make([]ident.Tag, 0, len(tags))
for _, t := range tags {
  identTags = append(identTags, ident.Tag{Name: ident.BytesID(t.Name), Value: ident.BytesID(t.Value)})
}
return ident.NewTagsIterator(ident.NewTags(identTags...)))
```

Reply:
Oh nice, don't know why I missed the `ident.Bytes`; was probably looking for something like `ident.ByteTag` or something :p